src/System.Private.Reflection.Core/src/System/Reflection/Runtime/TypeInfos/EcmaFormat/EcmaFormatRuntimeNamedTypeInfo.cs

```diff
@@ -0,0 +1,356 @@
+                            if (firstArg.Value == null)
+                                continue;
+
+                            string guidString = firstArg.Value as string;
+                            if (guidString == null)
+                                continue;
+
+                            return new Guid(guidString);
+                        }
+                    }
+                }
+            }
+
+            return null;
+        }
+
+        protected sealed override void GetPackSizeAndSize(out int packSize, out int size)
+        {
+            TypeLayout layout = _typeDefinition.GetLayout();
+            packSize = layout.PackingSize;
+            size = unchecked((int)(layout.Size));
+        }
+
+        public sealed override bool IsGenericTypeDefinition
+        {
+            get
+            {
+                return _typeDefinition.GetGenericParameters().Count != 0;
+            }
+        }
+
+        public sealed override string Namespace
+        {
+            get
+            {
+#if ENABLE_REFLECTION_TRACE
+                if (ReflectionTrace.Enabled)
+                    ReflectionTrace.TypeInfo_Namespace(this);
+#endif
+
+                return _reader.GetString(_typeDefinition.Namespace).EscapeTypeNameIdentifier();
+            }
+        }
+
+        public sealed override Type GetGenericTypeDefinition()
+        {
+            if (IsGenericTypeDefinition)
+                return this;
+            return base.GetGenericTypeDefinition();
+        }
+
+        public sealed override string ToString()
+        {
+            StringBuilder sb = null;
+
+            foreach (GenericParameterHandle genericParameterHandle in _typeDefinition.GetGenericParameters())
+            {
+                if (sb == null)
+                {
+                    sb = new StringBuilder(FullName);
+                    sb.Append('[');
+                }
+                else
+                {
+                    sb.Append(',');
+                }
+
+                GenericParameter genericParameter = _reader.GetGenericParameter(genericParameterHandle);
+                sb.Append(genericParameter.Name.GetString(_reader));
+            }
+
+            if (sb == null)
+            {
+                return FullName;
+            }
+            else
+            {
+                return sb.Append(']').ToString();
+            }
+        }
+
+        protected sealed override TypeAttributes GetAttributeFlagsImpl()
+        {
+            TypeAttributes attr = _typeDefinition.Attributes;
+            return attr;
+        }
+
+        protected sealed override int InternalGetHashCode()
+        {
+            return _typeDefinitionHandle.GetHashCode();
+        }
+
+        internal sealed override Type InternalDeclaringType
+        {
+            get
+            {
+                RuntimeTypeInfo declaringType = null;
+                if (((int)_typeDefinition.Attributes & 7) >= 2) // Check visibility mask on TypeDefinition
```

Comment:
Turns out that's how you check if a type is nested. ... But its a terrible check. I've switched it to use a helper function.